### ISSUE TITLE
Where Do I Belong Challenge Passing Requisites Not Checking

### ISSUE THREAD

**Issue by emanf9** [FreeCodeCamp/FreeCodeCamp#6560]
Challenge [Where do I belong](http://www.freecodecamp.com/challenges/where-do-i-belong#?solution=function%20where(arr%2C%20num)%20%7B%0A%20%20%2F%2F%20Find%20my%20place%20in%20this%20sorted%20array.%0A%20%20arr.push(num)%3B%0A%20%20arr.sort(function(a%2C%20b)%7Breturn%20a-b%3B%7D)%3B%0A%20%20var%20x%20%3D%200%3B%0A%20%20var%20result%20%3D%200%3B%0A%20%20for%20(x%20in%20arr)%7B%0A%20%20%20%20if%20(arr%5Bx%5D%20%3D%3D%20num)%7B%0A%20%20%20%20%20%20return%20x%3B%0A%20%20%20%20%7D%0A%20%20%20%20x%2B%2B%3B%0A%20%20%7D%0A%7D%0A%0Awhere(%5B10%2C%2020%2C%2030%2C%2040%2C%2050%5D%2C%2035)%3B) has an issue.
User Agent is: <code>Mozilla/5.0 (Windows NT 10.0; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/47.0.2526.111 Safari/537.36</code>.

So basically, my code always returns the right number for this challange, but it never completes the challenge. I checked all of the values it checks, and they all return the right value.  Any help?

To replicate, just paste the code below.

My code:
```javascript
function where(arr, num) {
  arr.push(num);
  arr.sort(function(a, b){return a-b;});
  var x = 0;
  var result = 0;
  for (x in arr){
    if (arr[x] == num){
      return x;
    }
    x++;
  }
}

where([10, 20, 30, 40, 50], 35);
```

**Comment by ltegman** [comment]
I can assure you you're not returning the right value. You're learning one of the many reasons you don't use `for in` loops to iterate over arrays. Please visit the [Help Chat](https://gitter.im/FreeCodeCamp/HelpJavaScript) for further assistance since Github issues are not for challenge help. Thanks and happy coding!

**Comment by ernest-okot** [comment]
Hello,

Am having a similar issue with this challenge

Here is my code

``` js
function where(arr, num) {
  return arr
    .sort(function(a, b) {
      return a > b;
    })
    .reduce(function(lowest, elm, index, list) {
      if (num <= elm) {
        console.log(index + ' --> new lowest index, lowest is now ' + lowest);
        return index;
      } else {
        console.log(index + ' --> same lowest index, lowest is now ' + lowest);
        return lowest;
      }
    }, 0);
}

where([10, 20, 30, 40, 50], 35);
```

That code returns `4` whilst in my opinion it's supposed to return `3`

Here is the console log

```
0 --> same lowest index, lowest is now 0
1 --> same lowest index, lowest is now 0
2 --> same lowest index, lowest is now 0
3 --> new lowest index, lowest is now 0
4 --> new lowest index, lowest is now 3
```

I get a feeling there's something wrong with the challenge
